### ISSUE TITLE
Ability to fake array of objects

### ISSUE THREAD

**Issue by sepehr500** [APIs-guru/graphql-faker#25]
I would like the ability to fake an array of objects. For example,
`@examples(values: [{jurisdiction: "DFT"},{jurisdiction: "FUR"}]) `

The source code made is seem like this is possible. How are you supposed to do it?

**Comment by IvanGoncharov** [comment]
@sepehr500 Just to be sure, you trying to write something like this?
```graphql
type Query {
  allCourtCases: [CourtCase!]!
     @examples(values: [{jurisdiction: "DFT"},{jurisdiction: "FUR"}])
}

type CourtCase {
  jurisdiction: String
}
```
And you want `allCourtCases` always return `[{jurisdiction: "DFT"},{jurisdiction: "FUR"}]`?

If you don't depend on the exact value you can write something like this:
```graphql
type CourtCase {
  jurisdiction: String @examples(values: ["DFT", "FUR"])
}
```

If above solution doesn't solve your problem can you please describe your use case in more details?

**Comment by sepehr500** [comment]
I am trying to do something like this: 

``` type Query {
  allCourtCases: [CourtCase!]!
     @examples(values: [{state: "WASH", capital: "SEA"},{state: "VA", capital: "RMD"}])
}

type CourtCase {
  capital: String
  state: String
}```

Sometimes  I need certain object properties to always appear together. For example, if I have city and a capital property, I may have logic that makes assumptions about about what other fields can be given some other field, so I would like to be able to mock an object. Otherwise, the API would return a random combination of values, and this does not always make sense.

**Comment by IvanGoncharov** [comment]
@sepehr500 I made `@examples` to work on objects.
It was pretty simple, however supporting an array of fixed size is way more complicated.
Not sure if it fully solves your issue, please see example below:
```graphql
type Query {
  allCourtCases: [CourtCase!]!
     @examples(values: [
       # if this object selected 'name' is always null
       {state: "WASH", capital: "SEA", name: null},
       # in this case 'name' is faked as usual
       {state: "VA", capital: "RMD"}
     ])
}

type CourtCase {
  name: String @fake(type:firstName)
  capital: String
  state: String
}
```
Example of possible result:
```json
{
  "data": {
    "allCourtCases": [
      {
        "name": null,
        "capital": "SEA",
        "state": "WASH"
      },
      {
        "name": "Bria",
        "capital": "RMD",
        "state": "VA"
      },
      {
        "name": null,
        "capital": "SEA",
        "state": "WASH"
      },
      {
        "name": "Guadalupe",
        "capital": "RMD",
        "state": "VA"
      }
    ]
  }
}
```

Does it solve your issue?
Or you always need an exact number of elements in the array?

**Comment by sepehr500** [comment]
When I was trying this out it was not working. Do you need to define every property on the object for it to work? Will the ones you don't define not fall through to the fake object property definitions?

**Comment by IvanGoncharov** [comment]
> When I was trying this out it was not working.

I pushed changes to [examplesOnObjects](https://github.com/APIs-guru/graphql-faker/tree/examplesOnObjects) a few minutes ago. You can try it there:
```sh
npm install
npm start
```

> Do you need to define every property on the object for it to work?

Only properties for which you want fixed values.

> Will the ones you don't define not fall through to the fake object property definitions?

By default, they will fall through and would be faked as usual to explicitly prevent this just define the property with `null` as the values.

**Comment by sepehr500** [comment]
Looks great! Have not had time to try it yet, but I had question. Given that `values' takes an array of objects that match the properties type, would it make more sense to pass in an array of objects into values like I did in my example? That way if you want to mock a single object, the same pattern of an array of objects would work.

**Comment by IvanGoncharov** [comment]
> Looks great! Have not had time to try it yet

@sepehr500 I pulished experimental NPM package with this change so you only need to run:
```sh
npm install -g graphql-faker@1.3.1-examplesOnObject
```
Can you please test it and If it solves your problem I will include this change into the next release.

> Given that `values' takes an array of objects that match the properties type, would it make more sense to pass in an array of objects into values like I did in my example? That way if you want to mock a single object, the same pattern of an array of objects would work.

Didn't fully understand your question can you please provide some examples?

**Comment by sepehr500** [comment]
Tried it out, and it works great! Let me rephrase my previous question

Currently, if want to fake a string, I do the following

```graphql
type CourtCase {
  jurisdiction: String @examples(values: ["DFT", "FUR"])
}
```
This makes the think that `values` expects it's value to be an array of whatever the properties type is. So it would follow, that if the type was the following,

```graphql
type CourtCase {
  jurisdiction: [String] @examples(values: [["DFT", "FUR"],["DFS","DNA"])
}
```
Since the property type is now an array of strings, it would make sense to pass in an array of strings into the array to match the return type. So if we take this same logic an apply it to Objects, then we would expect a property that has a type that is an array of objects, to look like the following,

```graphql
type Query {
  allCourtCases: [CourtCase!]!
     @examples(values: [[{state: "WASH", capital: "SEA"},{state: "VA", capital: "RMD"}], [{state:"VA"}]])
}
```
This way, you have a consistent rule that `values` always expects an array of whatever type the return type of the property is.

**Comment by IvanGoncharov** [comment]
@sepehr500 I released support of `@example` on objects in `1.4.0` 🎉 
Thank you for reporting this issue and highlighting inconsistency with `values` 👍
